modules/drivers/snowflake/src/metabase/driver/snowflake.clj

```diff
@@ -587,6 +591,28 @@
   [driver ps i t]
   (sql-jdbc.execute/set-parameter driver ps i (t/sql-timestamp (t/with-zone-same-instant t (t/zone-id "UTC")))))
 
+;;; --------------------------------------------------- Query remarks ---------------------------------------------------
+
+;  Snowflake strips comments prepended to the SQL statement (default remark injection behavior). We should append the
+;  remark instead.
+(defmethod sql-jdbc.execute/inject-remark :snowflake
+  [_ sql remark]
+  (str sql "\n\n-- " remark))
+
+(defmethod qp.util/query->remark :snowflake
+  [_ {{:keys [context executed-by action-id card-id pulse-id dashboard-id database-id query-hash]} :info,
+      query-type :type}]
+  (json/generate-string {:client      "Metabase"
+                         :context     context
+                         :queryType   query-type
+                         :userId      executed-by
+                         :actionId    action-id
+                         :pulseId     pulse-id
+                         :cardId      card-id
+                         :dashboardId dashboard-id
+                         :databaseId  database-id
+                         :queryHash   (if (bytes? query-hash) (codecs/bytes->hex query-hash) nil)
```

Comment:
This can be even further simplified:
```suggestion
                         :queryHash   (when (bytes? query-hash) (codecs/bytes->hex query-hash))
```

Reply:
Thanks! I've updated the code with your suggestion.

Just for the sake of documentation: I was curious and checked what would happen if we remove the check for `bytes?` altogether. It causes for example table fingerprinting to fail with:

```
2023-11-26 14:28:34,024 WARN sync.util :: Error fingerprinting Table 81 '[redacted]'
java.lang.NullPointerException
        at org.apache.commons.codec.binary.Hex.encodeHex(Hex.java:171)
        at org.apache.commons.codec.binary.Hex.encodeHex(Hex.java:156)
        at org.apache.commons.codec.binary.Hex.encodeHex(Hex.java:142)
        at org.apache.commons.codec.binary.Hex.encodeHexString(Hex.java:290)
        ...
```